### DISCUSSION TITLE
Check formula for errors

### DISCUSSION THREAD

**Discussion by Bykiev** [AntonovAnton/math.evaluation#42] [Q&A] [answered]
Hi, is it possible to check the formula for errors before evaluation? For ex: 
- parentheses or braces are not closed
- invalid tokens
and get it's position?

**Comment by AntonovAnton** [comment]
Hi! Please provide examples of expressions and what you are expecting to get. Currently the evaluator throws NotSupportedException with details what part of the expression it doesn't recognize. Also it throws FormatException if it doesn't recognize numbers.

**Reply by Bykiev** [reply]
"12 + 3 * (120 +5"
"12 + sin()"
"12 + sin + 1"
and etc

**Reply by AntonovAnton** [reply, marked as answer]
Hi! I’ve added error checking in [version 1.1.8](https://github.com/AntonovAnton/math.evaluation/releases/tag/1.1.8) for cases such as:

"12 + 3 * (120 + 5" (missing closing parenthesis)
"12 + sin()" (empty function call)
"12 + sin / 1" (incomplete function usage)
However, expressions like "12 + sin + 1" are still evaluated, as they can be interpreted as valid mathematical expressions, such as "12 + sin(+1)" by some evaluators. I’m avoiding strict restrictions to maintain flexibility in cases like "cos -1" and other similar scenarios.

**Reply by AntonovAnton** [reply]
Hi! I double-checked, and the minus sign (-) is interpreted as a negative sign, whereas the plus sign (+) is generally interpreted as an addition operation. Therefore, I’ve decided to add error checking in [version 1.1.9](https://github.com/AntonovAnton/math.evaluation/releases/tag/1.1.9). As a result, expressions like "sin + 1" will now be considered invalid due to incomplete function usage, while "sin - 1" will still be valid.

For example, the expression "12 + sin + 1" will now throw an exception:
"Error evaluating the expression. The operand is not recognizable. Invalid token at position 8."
